requests/sessions.py

```diff
@@ -239,10 +239,10 @@ def rebuild_auth(self, prepared_request, response):
         if 'Authorization' in headers:
             # If we get redirected to a new host, we should strip out any
             # authentication headers.
-            original_parsed = urlparse(response.request.url)
-            redirect_parsed = urlparse(url)
+            original_root = urljoin(response.request.url, '/')
```

Comment:
We may want to do the explicit check on scheme, hostname, and port here. Otherwise a url with credentials that’s redirected without credentials may have the Authorization header unnecessarily stripped.

Reply:
Fair enough, I hadn't spotted that the username and password are still present in response.request.url.